### PR TITLE
[TML-75] Add title fix button

### PR DESCRIPTION
**Pull request by vazish**
## Goal
Based on meeting we had with Editorial team, we wanted to add a button that fixes the title and the straight apostrophe's in titles when populating from a url.

Created two helper functions for these two fixes:

Add curly quotes and fix AP style title case for titles in collections (see tests also)

## Reference

Documentation here:
1. Casing follows AP style here - https://headlinecapitalization.com/
2. Regular single and double quotes replaced by curly ones

Tickets:

[Jira](https://getpocket.atlassian.net/jira/software/projects/TML/boards/123?assignee=637ba9dca593cb822e93c33f)
Screenshot 2023-06-15 at 12 27 14

### CODE REVIEW

File: src/_shared/utils/applyCurlyQuotes.ts [hunk 1]
```diff
@@ -0,0 +1,15 @@
+/**
+ * Helper to replace opening and closing curly single and double quotes
+ *
+ * @param text
+ * @returns {string}
+ */
+
+export const applyCurlyQuotes = (text: string): string => {
+  if (!text) return '';
```
Comment (Herraj) [inline src/_shared/utils/applyCurlyQuotes.ts:9]: nit pick: We try not to omit the `{}` for a if block even if it is a single line 😅

File: src/_shared/utils/applyApTitleCase.ts [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+        index === 0 ||
+        index === all.length - 1 ||
+        !stop.includes(word.toLowerCase())
+      ) {
+        return capitalize(word);
+      }
+      return word.toLowerCase();
+    })
+    .join('');
+};
```
Comment (Herraj) [inline src/_shared/utils/applyApTitleCase.ts:43]: This looks good! I would just add a small block comment at the very top mentioning the library we tried to use and that this a custom implementation of it, for reference purposes.

File: src/_shared/utils/applyApTitleCase.test.ts [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+    const articles = [
+      {
+        result: 'Ernest Hemingway wrote For Whom The Bell Tolls.',
+        expected: 'Ernest Hemingway Wrote for Whom the Bell Tolls.',
+      },
+      {
+        result: 'Girl on A Train is a thriller by A. J. Waines.',
+        expected: 'Girl on a Train Is a Thriller by A. J. Waines.',
+      },
+    ];
```
Comment (Herraj) [inline src/_shared/utils/applyApTitleCase.test.ts:23]: Nice!

**Review by Herraj — APPROVED** [review]

File: src/_shared/utils/applyApTitleCase.ts (hunk 2 above)
Comment (vazish) [reply src/_shared/utils/applyApTitleCase.ts:43]: Yup I added in the function description

File: src/_shared/utils/applyApTitleCase.ts [hunk 4]
```diff
@@ -10,20 +10,24 @@ export const SEPARATORS = /(\s+|[-‑–—,:;!?()])/;
...
+    return '';
+  }
   return value.charAt(0).toUpperCase() + value.slice(1);
 };
 
 /**
  * Helper to convert text to AP title case
  * adapted from https://github.com/words/ap-style-title-case
- * should match https://headlinecapitalization.com/
+ * text should match https://headlinecapitalization.com/
```
Comment (Herraj) [inline src/_shared/utils/applyApTitleCase.ts:22]: whoops looks like you had it already and I missed it.

**Review by Herraj — APPROVED** [review]

**Pull request closed without merging**